tests/ignite/engine/test_create_supervised.py

```diff
@@ -93,7 +102,9 @@ def _test_create_supervised_evaluator(
         example_input = torch.randn(1, 1)
         model = torch.jit.trace(model, example_input)
 
-    evaluator = create_supervised_evaluator(model, device=evaluator_device)
+    evaluator = create_supervised_evaluator(model, device=evaluator_device, amp_mode=amp_mode)
+
+    assert evaluator._process_function is supervised_evaluation_step or supervised_evaluation_step_amp
```

Comment:
Well, the point here is to assert the function depending on `amp_mode` as it is always one of those 2 functions.

Reply:
ah ok my bad. changing now.

Reply:
@vfdev-5  also I feel that it's more standard to use `assert_called` instead of using the `process_function`. 
I feel that using `process_function` defies the point of the test as it requires us to know how the implementation is. while the unittest should just test the function behavior. what do you think?

Reply:
Yes, that's true and asserting if it was called is also a correct wat to do that but requires execution. On the other hand, here we test the function that sets up an engine and to go fast we just assert that it did what we'd like. I think it is OK for now here.

Reply:
According the the CI failure, we do not test it as we intend to. Failure is correct that process function is not the function used to create the wrapper... Probably, there is no way to check it ...

Reply:
Yes I guess that the best way for now is to remove that assertion. I am not sure testing that the method was called is possible without mocking after execution

Reply:
Sounds good! Let's think about how to mock things in other issues/PRs. If you'd like to make a progress on that, let's expose the ideas in an issue and discuss there about what would be interesting to do.